controllers/daemonset_reconcile.go

```diff
@@ -144,10 +154,315 @@ func (r *KataConfigOpenShiftReconciler) daemonSetDeployment() (ctrl.Result, erro
 }
 
 func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequestDaemonSet() (ctrl.Result, error) {
-	// TODO: Remove packages, configs from nodes, remove labels, remove finalizer
+	r.Log.Info("KataConfig deletion in progress: ")
+
+	if contains(r.kataConfig.GetFinalizers(), kataConfigFinalizer) {
+		// Get the list of pods that might be running using kata runtime
+		err := r.listKataPods()
+		if err != nil {
+			r.setInProgressConditionToBlockedByExistingKataPods(err.Error())
+			updErr := r.Client.Status().Update(context.TODO(), r.kataConfig)
+			if updErr != nil {
+				return ctrl.Result{}, updErr
+			}
+			r.Log.Info("Kata PODs are present. Requeue for reconciliation ")
+			return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
+		}
+	}
+
+	r.setInProgressConditionToUninstalling()
+
+	// Delete the kata install ds if it exists
+	kataInstallDs, err := r.daemonSetForKataInstall(InstallKata)
+	if err != nil {
+		r.Log.Error(err, "Failed getting DaemonSet for Kata installation")
+		return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
+	}
+	err = r.Client.Delete(context.TODO(), kataInstallDs)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.Log.Info("kata install daemonset was already deleted")
+		} else {
+			r.Log.Error(err, "error when deleting kata install Daemonset, try again")
+			return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 15}, err
+		}
+	}
+
+	isUninstallDaemonSetJustCreated := false
+
+	// Create kata uninstall ds if it doesn't exist
+	// Save whether it was just created or not
+	kataUninstallDs, err := r.daemonSetForKataInstall(UninstallKata)
+	if err != nil {
+		r.Log.Error(err, "Failed getting DaemonSet for Kata uninstallation")
+		return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
+	}
+	if err := controllerutil.SetControllerReference(r.kataConfig, kataUninstallDs, r.Scheme); err != nil {
+		r.Log.Error(err, "Failed setting ControllerReference for kata uninstallation DS")
+		return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
+	}
+
+	foundKataDs := &appsv1.DaemonSet{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: kataUninstallDs.Name, Namespace: kataUninstallDs.Namespace}, foundKataDs)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.Log.Info("Creating a new kata uninstallation daemonset", "kataUninstallDs.Namespace", kataUninstallDs.Namespace, "kataUninstallDs.Name", kataUninstallDs.Name)
+			err = r.Client.Create(context.TODO(), kataUninstallDs)
+			if err != nil {
+				r.Log.Error(err, "error when creating kata uninstallation daemonset")
+				return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, err
+			}
+			isUninstallDaemonSetJustCreated = true
+
+			kataSelector, err := r.getKataConfigNodeSelectorAsSelector()
+			if err != nil {
+				r.Log.Error(err, "could not get kata node selector, skipping waiting to uninstall labeling")
```

Comment:
Sorry, I really have trouble parsing this error. What does "skipping waiting" mean in that context? Do you mean "skipping" waiting or "skipping to some later step", or "waiting for (labelling)", or some combination of the above?

```suggestion
				r.Log.Error(err, "could not get kata node selector, will skip labelling of nodes to uninstall")uninstall labelling")
```